Sources/Plasma/CoreLib/hsStream.h

```diff
@@ -126,19 +126,30 @@ enum {
     void            WriteLEDouble(double value);
     void            WriteLEDouble(size_t count, const double values[]);
 
-    template <typename T>
-    inline void ReadLE(T* value)
-    {
-        Read(sizeof(T), value);
-        *value = hsToLE(*value);
-    }
-
-    template <typename T>
-    inline void WriteLE(T value)
-    {
-        value = hsToLE(value);
-        Write(sizeof(T), &value);
-    }
+    // Type-safe placement readers
+    template <typename T> inline void ReadByte(T*) = delete;
+    void ReadByte(uint8_t* v) { *v = ReadByte(); }
+    void ReadByte(int8_t* v) { *v = (int8_t)ReadByte(); }
+    template <typename T> inline void ReadLE16(T*) = delete;
+    void ReadLE16(uint16_t* v) { *v = ReadLE16(); }
+    void ReadLE16(int16_t* v) { *v = (int16_t)ReadLE16(); }
+    template <typename T> inline void ReadLE32(T*) = delete;
+    void ReadLE32(uint32_t* v) { *v = ReadLE32(); }
+    void ReadLE32(int32_t* v) { *v = (int32_t)ReadLE32(); }
+    template <typename T> inline void ReadLEFloat(T*) = delete;
+    void ReadLEFloat(float* v) { *v = ReadLEFloat(); }
+    template <typename T> inline void ReadLEDouble(T*) = delete;
+    void ReadLEDouble(double* v) { *v = ReadLEDouble(); }
```

Comment:
I wonder if these should be marked `[[nodiscard]]`? I'm assuming the reason for the `(void)` when reading and throwing away with `(void)s->ReadLE32();` is to silence warnings about return values?

Reply:
Yeah, I actually did have `[[nodiscard]]` at one point (hence the `(void)` casts), but I later changed my mind...  I could add them back in if preferred.

Reply:
https://www.youtube.com/watch?v=CxK_nA2iVXw